homeassistant/components/opensprinkler/sensor.py

```diff
@@ -0,0 +1,177 @@
+"""Opensprinkler integration."""
+import logging
+from typing import Callable
+
+from homeassistant.const import CONF_NAME, DEVICE_CLASS_TIMESTAMP
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity
+from homeassistant.util.dt import utc_from_timestamp
+
+from . import OpensprinklerCoordinator, OpensprinklerSensor
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: dict, async_add_entities: Callable,
+):
+    """Set up the opensprinkler sensors."""
+    entities = _create_entities(hass, entry)
+    async_add_entities(entities)
+
+
+def _create_entities(hass: HomeAssistant, entry: dict):
+    entities = []
+
+    device = hass.data[DOMAIN][entry.entry_id]
+    name = entry.data[CONF_NAME]
+    coordinator = OpensprinklerCoordinator(hass, device)
```

Comment:
I think this makes more sense to attach to the controller instance and then leverage the same coordinator wherever it may be used in the future.

Reply:
Yes it should be looked at as we add more platforms